fuzz/fuzz_targets/streams.rs

```diff
@@ -0,0 +1,68 @@
+#![no_main]
+
+use arbitrary::{Arbitrary, Unstructured};
+use libfuzzer_sys::fuzz_target;
+
+extern crate proto;
+use proto::fuzzing::{ResetStream, Streams};
+use proto::{Dir, Side, StreamId, VarInt};
+
+#[derive(Arbitrary, Debug)]
+struct StreamParams {
+    side: Side,
+    max_remote_uni: u16,
+    max_remote_bi: u16,
+    send_window: u16,
+    receive_window: u16,
+    stream_receive_window: u16,
+    dir: Dir,
+}
+
+#[derive(Arbitrary, Debug)]
+enum Operation
+{
+    Open(Vec<u8>),
+    Accept(Dir),
+    Finish(StreamId),
+    ReceivedStopSending(StreamId, VarInt),
+    ReceivedReset(ResetStream),
+    Reset(StreamId),
+}
+
+fuzz_target!(|input: (StreamParams, Vec<Operation>)| {
+    let (params, operations) = input;
+    let mut stream = Streams::new(
+        params.side,
+        params.max_remote_uni.into(),
+        params.max_remote_bi.into(),
+        params.send_window.into(),
+        params.receive_window.into(),
+        params.stream_receive_window.into(),
+    );
+
+    for operation in operations {
+        match operation {
+            Operation::Open(v) => {
+                let mut data = Unstructured::new(&v[..]);
+                if let Ok(tp) = data.arbitrary() {
```

Comment:
I see that `v: Vec<u8>` here is only used to invent a `TransportParameters`. Could we avoid going through raw bytes here, and get the `TransportParameters` directly?

Also, `Streams` isn't expected to handle dynamically changing `TransportParameters`, so it would probably make more sense to generate it once via `StreamParams` and reference it as needed, rather than coming up with a fresh one on every `Open`.

Reply:
yep this makes sense, good to know. 

I've squash and rebased all but the last commit that drops the raw bytes and `TransportParameters` only once rather than each call to open. If this is the right track I will squash it in with the rest.